### PR TITLE
MOTECH-1953: Add a period widget to the tasks module

### PR DESCRIPTION
**Pull request by 1stmateusz**
I added period widget from mds to tasks module and moved directive from mds
to server-bundle

### CODE REVIEW

File: modules/tasks/tasks/src/main/resources/webapp/js/controllers.js [hunk 1]
```diff
@@ -1382,6 +1382,10 @@
                 return message;
             }
         };
+
+        $scope.closePeriodModal = function () {
```
Comment (ngraczewski) [inline modules/tasks/tasks/src/main/resources/webapp/js/controllers.js]: Why don't you use data-dismiss attribute on the close button of the modal?

File: modules/tasks/tasks/src/main/resources/webapp/js/directives.js [hunk 2]
```diff
@@ -1386,4 +1386,207 @@
         };
     });
 
+    directives.directive('periodAmount', function() {
```
Comment (ngraczewski) [inline modules/tasks/tasks/src/main/resources/webapp/js/directives.js]: Isn't this already in the server-bundle?

File: modules/tasks/tasks/src/main/resources/webapp/partials/widgets/content-editable-period.html [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                    <label class="slider-label" for="amount-period-hour">{{msg('task.dateTimeUnits.hours')}}&#58;</label><input type="text" id="amount-period-hour" readonly>
+                    <div id="period-hour">0</div>
+                    <label class="slider-label" for="amount-period-minute">{{msg('task.dateTimeUnits.minutes')}}&#58;</label><input type="text" id="amount-period-minute" readonly>
+                    <div id="period-minute">0</div>
+                    <label class="slider-label" for="amount-period-second">{{msg('task.dateTimeUnits.seconds')}}&#58;</label><input type="text" id="amount-period-second" readonly>
+                    <div id="period-second">0</div>
+                </div>
+            </div>
+            <div class="modal-footer">
+                <button class="btn btn-default" ng-click="closePeriodModal()">{{msg('task.button.close')}}</button>
```
Comment (ngraczewski) [inline modules/tasks/tasks/src/main/resources/webapp/partials/widgets/content-editable-period.html]: Why don't you use the data-dismiss attribute here? Any reason for calling the function?

File: modules/tasks/tasks/src/main/resources/webapp/partials/widgets/content-editable-period.html [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    {{msg('task.btn.changePeriod')}}
+</button>
+<div contenteditable ng-model="data.value" droppable class="actionField period-input period-field word-wrap input-extra-fluid" data-index="{{index}}" data-type="{{data.type}}">
+    <span ng-bind-html-unsafe="data.value" ng-model="data.value" class="html">
+        <h1 contentEditable="false">
+            <span class="editable"></span>
+        </h1>
+    </span>
+</div>
+<div id="periodModal" class="tasks modal fade" tabindex="-1" role="dialog" aria-hidden="true" data-index="{{index}}" data-action="{{action}}">
```
Comment (ngraczewski) [inline modules/tasks/tasks/src/main/resources/webapp/partials/widgets/content-editable-period.html]: This div and the under [this link](https://github.com/motech/motech/blob/master/platform/mds/mds-web/src/main/resources/webapp/partials/widgets/field-basic-defaultValue-period.html#L7) are basically the same. What about placing this code in a separate .html file and loading it via angular directive? Here is a working example - [link](https://github.com/motech/motech/blob/master/platform/mds/mds-web/src/main/resources/webapp/js/directives.js#L3363).

**Comment by pgesek** [comment]
test this please

**Comment by ngraczewski** [comment]
test this please

**Comment by ngraczewski** [comment]
test this please

**Comment by pgesek** [comment]
test this please

**Comment by pgesek** [comment]
test this please

**Comment by pgesek** [comment]
test this please

**Comment by 1stmateusz** [comment]
test this please

**Comment by 1stmateusz** [comment]
test this please

**Pull request merged**